src/main/kotlin/it/pagopa/wallet/services/MigrationService.kt

```diff
@@ -29,21 +32,41 @@ import reactor.kotlin.core.publisher.toMono
 class MigrationService(
     private val walletPaymentManagerRepository: WalletPaymentManagerRepository,
     private val walletRepository: WalletRepository,
+    private val applicationRepository: ApplicationRepository,
     private val loggingEventRepository: LoggingEventRepository,
     private val uniqueIdUtils: UniqueIdUtils,
-    walletMigrationConfig: WalletMigrationConfig
+    private val walletMigrationConfig: WalletMigrationConfig
 ) {
 
     private val cardPaymentMethodId =
         PaymentMethodId(UUID.fromString(walletMigrationConfig.cardPaymentMethodId))
     private val logger: Logger = LoggerFactory.getLogger(this.javaClass)
 
+    private val pagoPaApplication by lazy {
+        applicationRepository
+            .findById(walletMigrationConfig.defaultApplicationId)
+            .map {
+                WalletApplication(
+                    WalletApplicationId(it.id),
+                    WalletApplicationStatus.ENABLED,
+                    Instant.now(),
+                    Instant.now(),
+                    WalletApplicationMetadata.empty()
+                )
+            }
```

Comment:
Doesn't this mean that each wallet migrated during the same day will have the same creation and update field on the wallet application? This would break the invariant that `application.creationDate >= wallet.creationDate` for each wallet and its application.
An alternative would be to just cache the `Application` and construct it while migrating

```suggestion
```

Reply:
`application.creationDate >= wallet.creationDate` yes, you're right, I will cache just the application and the update its properties during migration

Reply:
Fixed in [dc29d8b](https://github.com/pagopa/pagopa-wallet-service/pull/119/commits/dc29d8bcbf07d448027f863a3c0b6bdd03fd9550)